From 284afdc92eb2edffab500eb3768a9347bc5e77c8 Mon Sep 17 00:00:00 2001
From: Bernd Wachter <bwachter-tmw@lart.info>
Date: Sat, 23 Jan 2010 23:24:05 +0100
Subject: [PATCH] Allow version override with cmake parameters

---
 CMakeLists.txt |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4d9238d..dd10b71 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,7 +2,9 @@ CMAKE_MINIMUM_REQUIRED(VERSION 2.6)
 
 PROJECT(MANA)
 
-SET(VERSION 0.0.29.1)
+IF (NOT VERSION)
+    SET(VERSION 0.0.29.1)
+ENDIF()
 
 # where to look for cmake modules
 SET(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/CMake/Modules)
-- 
1.5.6.5

