From 5cc5f522cafbc892d695883660d891b24b9a4f6e Mon Sep 17 00:00:00 2001
From: Bernd Wachter <bwachter-tmw@lart.info>
Date: Sat, 23 Jan 2010 22:40:29 +0100
Subject: [PATCH] Add flag to actually enable gettext when building with ENABLE_NLS

---
 src/CMakeLists.txt |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index c06f7d9..ea04fc5 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -14,6 +14,10 @@ SET(FLAGS "-DPACKAGE_VERSION=\\\"${VERSION}\\\"")
 SET(FLAGS "${FLAGS} -DPKG_DATADIR=\\\"${PKG_DATADIR}/\\\"")
 SET(FLAGS "${FLAGS} -DLOCALEDIR=\\\"${LOCALEDIR}/\\\"")
 
+IF (ENABLE_NLS)
+  SET(FLAGS "${FLAGS} -DENABLE_NLS=1")
+ENDIF()
+
 IF (WIN32)
     SET(EXTRA_LIBRARIES ws2_32 winmm)
 ELSEIF (CMAKE_SYSTEM_NAME STREQUAL SunOS)
-- 
1.5.6.5

